src/Engine.py

```diff
@@ -79,19 +89,18 @@ def __invokeTools(self, schedule):
 
     #invokes the tool via provided command (absolute path)
     def __invokeTool(self, task):  
-        Utils.printNotiMessage("RUNNING " + task.getToolName() + "...") 
-        commands = task.getCommand()  
+        commands = task.command  
```

Comment:
Same here. In my personal opinion, using getCommand() is better as by doing task.command someone can use task.command = blahblah to ruin the parameters of `task`.

Reply:
Yea originally, for the reasons you mentioned, I made the variables only read accessible through the functions when the task object only had two variables, but later I decided for the task object to hold more data (8 variables), so it was either that I create get methods for each of these variables or make them fully accessible. I was under the impression that creating several  get functions would make the code more cluttered so decided to just make them fully accessible, but I am not sure if it was the best choice. If you believe that using get methods and making the variables private instead is better, then I can change it back to that.

Reply:
I'll switch it back later. I want to focus on some functional aspects first.